app/models/user.rb

```diff
@@ -36,6 +36,7 @@ class User < ApplicationRecord
 
   SEARCH_SERIALIZER = Search::UserSerializer
   SEARCH_CLASS = Search::User
+  DATA_SYNC_CLASS = DataSync::Elasticsearch::User
```

Comment:
I'm a bit torn here. On the one hand I' m always careful with adding too much magic, but on the other hand I'm also not a big fan of adding constants for class names that can be inferred programatically. For example, this worked fine for me locally:

```ruby
  def sync_related_elasticsearch_docs
    data_sync_class = DataSync::Elasticsearch.const_get(self.class.name)
    data_sync_class.new(self, saved_changes).sync_documents
  end
```

What I usually do is wrapping these things up behind methods:

```ruby
def data_sync_class
  DataSync::Elasticsearch.const_get(self.class.name)
end

def sync_related_elasticsearch_docs
  data_sync_class.new(self, saved_changes).sync_documents
end
```

So if a class really needs a different name that is not so easily inferred, it can just override the relevant method, but the majority of classes where the naming is predictable don't need to do anything. 

Something similar would work for `SEARCH_CLASS` and `SEARCH_SERIALIZER`. We don't have to do this now though.

Reply:
Oh, that is a good thought. I stuck with this since it was the convention we were following but I do like not saving classes in constants.